main.go

```diff
@@ -16,17 +19,53 @@ func proxyHandler(w http.ResponseWriter, r *http.Request) {
 		"header":     r.Header,
 	}).Info("Received /proxy request")
 
+	target := r.Header.Get("X-OXYproxy")
+	if target == "" {
+		log.Warning("Target source is undefined")
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte("500 - Something bad happened! Destination source is not set..."))
+	}
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Println("Error reading body", err)
+		http.Error(w, "can't read body", http.StatusBadRequest)
+		return
+	}
+
+	proxyReq := &HTTPReq{
+		Method:     r.Method,
+		Proto:      r.Proto,
+		Header:     r.Header,
+		Body:       string(body),
+		Host:       r.Host,
+		Form:       r.Form,
+		Trailer:    r.Trailer,
+		RemoteAddr: r.RemoteAddr,
+		Target:     target,
+	}
+
+	jsonReq, err := json.Marshal(proxyReq)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
are you sure that in the error case you still should continue the workflow?
seems like you have to add return here

Reply:
I’m sure that we don’t need to continue, but in error case log.Fatal logs the error and exit program. Isn’t it? Why do we need do add also return if the program will be terminated because of log.Fatal